src/ConnectedMode/ConnectedModePackage.cs

```diff
@@ -65,6 +67,7 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
             boundSolutionUpdateHandler = componentModel.GetService<BoundSolutionUpdateHandler>();
             timedUpdateHandler = componentModel.GetService<TimedUpdateHandler>();
             localSuppressionsChangedHandler = componentModel.GetService<LocalSuppressionsChangedHandler>();
+            preSolutionBoundUpdateHandler = componentModel.GetService<PreSolutionBoundUpdateHandler>();
```

Comment:
Seeing the name of the new class here (`PreSolutionBoundUpdateHandler`) made me realise it doesn't give any real idea what the class is for. It's named after the event it listens to, rather than telling us what it does. We followed a similar naming pattern in when doing the original `Education` (e.g. `BoundSolutionUpdateHandler`, `TimedUpdateHandler`), and they suffer from the same problem.

I think it would be better if the name reflected what the class does - the fact that it handles a particular event   is an implementation detail. The problem then is coming up with a good name... `ImportBeforeInstallTrigger`? `CheckImportBeforeInstalledTrigger`? `TriggerImportBeforeCheckListener`?